### PR TITLE
Remove `Kubernetes` and `KubernetesConfiguration` from required resource providers list.

### LINKED ISSUE #22515: Breaking change - AzureRM provider 3.65 forces `Microsoft.Kubernetes` and `Microsoft.KubernetesConfiguration` resource provider registration
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Terraform Version

1.x

### AzureRM Provider Version

3.65

### Affected Resource(s)/Data Source(s)

Any

### Terraform Configuration Files

```hcl
Any
```

### Debug Output/Panic Output

```shell

│ Original Error: Cannot register providers: Microsoft.Kubernetes, Microsoft.KubernetesConfiguration. Errors were: Cannot register provider Microsoft.Kubernetes with Azure Resource Manager: providers.ProvidersClient#Register: Failure responding to request: StatusCode=403 -- Original Error: autorest/azure: Service returned an error. Status=403 Code="AuthorizationFailed" Message="The client 'redacted' with object id 'redacted' does not have authorization to perform action 'Microsoft.Kubernetes/register/action' over scope '/subscriptions/redacted' or the scope is invalid. If access was recently granted, please refresh your credentials.".
│ Cannot register provider Microsoft.KubernetesConfiguration with Azure Resource Manager: providers.ProvidersClient#Register: Failure responding to request: StatusCode=403 -- Original Error: autorest/azure: Service returned an error. Status=403 Code="AuthorizationFailed" Message="The client 'redacted' with object id 'redacted' does not have authorization to perform action 'Microsoft.KubernetesConfiguration/register/action' over scope '/subscriptions/redacted' or the scope is invalid. If access was recently granted, please refresh your credentials.".
│ 
│   with provider["registry.terraform.io/hashicorp/azurerm"],
│   on provider.tf line 3, in provider "azurerm":
│    3: provider "azurerm" {

```

### Expected Behaviour

Resource providers should not be force registered. 

### Actual Behaviour

`Microsoft.Kubernetes` and `Microsoft.KubernetesConfiguration` were added as a required resource provider in AzureRM 3.65.0. However, it's not a mandatory resource, and is a breaking change.

[Microsoft's recommendation is to](https://learn.microsoft.com/en-us/azure/azure-resource-manager/management/azure-services-resource-providers#registration) "Only register a resource provider when you're ready to use it. The registration step enables you to maintain least privileges within your subscription. A malicious user can't use resource providers that aren't registered.".

The AzureRM provider requiring registration of a non-mainstream resource is not best practice.

The release notes for the AzureRM provider release 3.65.0 do not mention that a new resource registration is required.

### Steps to Reproduce

Deploy any AzureRM resource (except Azure Kubernetes) to a subscription without the Kubernetes resource provider registered, using a service connection scoped at any level other than the subscription.
The deployment fails with the error.

### Important Factoids

This also happened in provider version 3.51 - see [this 21363](https://github.com/hashicorp/terraform-provider-azurerm/issues/21363)

### References

https://github.com/hashicorp/terraform-provider-azurerm/pull/22463
https://github.com/hashicorp/terraform-provider-azurerm/issues/22462
https://github.com/hashicorp/terraform-provider-azurerm/issues/21363

### PR DESCRIPTION
**Pull request by stephybun**
Closes #22515

### CODE REVIEW

File: website/docs/index.html.markdown [hunk 1]
```diff
@@ -45,7 +47,9 @@ terraform {
 
 # Configure the Microsoft Azure Provider
 provider "azurerm" {
-  features {}
+  features {
+    skip_provider_registration = true # This is only required when the user running Terraform lacks the permissions to register Azure Resource Providers.
```
Comment (jackofallops) [inline website/docs/index.html.markdown:51]: I think this should be a separate example, and we should keep the basic example also, but update the comment to be explicit in the requirement for any auth method? 
```suggestion
    skip_provider_registration = true # This is only required when the User, Service Principal, or Identity running Terraform lacks the permissions to register Azure Resource Providers.
```

**Review by jackofallops — APPROVED** [review]
Thanks @stephybun - One suggestion, but otherwise LGTM 👍

File: website/docs/index.html.markdown [hunk 2]
```diff
@@ -29,6 +29,8 @@ Terraform supports a number of different methods for authenticating to Azure:
 
 We recommend using either a Service Principal or Managed Service Identity when running Terraform non-interactively (such as when running Terraform in a CI server) - and authenticating using the Azure CLI when running Terraform locally.
 
+->**Note:** The Service Principal or Managed Service Identity running Terraform should have permissions to register [Azure Resource Providers](https://learn.microsoft.com/en-us/azure/azure-resource-manager/management/resource-providers-and-types). If the user running Terraform has insufficient permissions to register Resource Providers then we recommend setting the property [skip_provider_registration](#skip_provider_registration) in the provider block to prevent auto-registration.
```
Comment (manicminer) [inline website/docs/index.html.markdown:32]: Minor but in addition to the above, I'd also shorten this to `Managed Identity` since that's the latest nomenclature we are striving towards
```suggestion
->**Note:** The User, Service Principal or Managed Identity running Terraform should have permissions to register [Azure Resource Providers](https://learn.microsoft.com/en-us/azure/azure-resource-manager/management/resource-providers-and-types). If the principal running Terraform has insufficient permissions to register Resource Providers then we recommend setting the property [skip_provider_registration](#skip_provider_registration) in the provider block to prevent auto-registration.
```

**Review by manicminer — APPROVED** [review]

File: website/docs/index.html.markdown (hunk 2 above)
Comment (stephybun) [reply website/docs/index.html.markdown:32]: Thanks @manicminer!

Comment (WodansSon) [inline website/docs/index.html.markdown]:
```suggestion
->**NOTE:** The Service Principal or Managed Service Identity running Terraform should have permissions to register [Azure Resource Providers](https://learn.microsoft.com/en-us/azure/azure-resource-manager/management/resource-providers-and-types). If the user running Terraform has insufficient permissions to register Resource Providers then we recommend setting the property [skip_provider_registration](#skip_provider_registration) in the provider block to prevent auto-registration.
```

Not sure if there is a Hashi guideline/best practice for this, but thought I would mention this super minor documentation inconsistency here. Should **Note:** be, **NOTE:**, **Note:** or **note:**?

**Review by WodansSon — APPROVED** [review]

File: website/docs/index.html.markdown (hunk 2 above)
Comment (stephybun) [reply website/docs/index.html.markdown]: AFAIK there's no guidelines and it's wildly inconsistent across all the docs - I personally go for **Note:** and in resources that I take care of I try be consistent with what's already in there. Uppercasing this to **NOTE:** would introduce an inconsistency to this particular page in the docs since all the other Notes are not uppercased.
